.github/workflows/build.yml

```diff
@@ -0,0 +1,56 @@
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Image version
+        required: true
+        type: string
+      run_tests:
+        description: Run tests
+        required: true
+        type: bool
+        default: true
+
+permissions:
+  packages: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: pnpm/action-setup@v3
+        with:
+          version: 8
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: 'pnpm'
+
+      - name: Install Cartesi CLI
+        run: pnpm add -g @cartesi/cli@0.14.1
+
+      - name: Build bounty examples
+        run: make bounties
+
+      - name: Set image tag
+        run: echo "TAG=ghcr.io/crypto-bug-hunters/bug-buster-machine:${{ inputs.version }}" >> "$GITHUB_ENV"
+
+      - name: Build the machine image with Docker
+        run: docker buildx build -t "$TAG" .
+
+      - name: Build the application with Cartesi CLI
+        run: cartesi build --from-image "$TAG"
+
+      - name: Build test image
+        if: ${{ inputs.run_tests }}
+        run: make test-image
+
+      - name: Run tests
+        if: ${{ inputs.run_tests }}
+        run: make test
+
+      - name: Push the machine image to GHCR
+        run: docker push "$TAG"
```

Comment:
The docker-push-action will make the push, but the login is needed before.

There's a docker/login-action at 

References:

- https://github.com/docker/login-action
- https://github.com/cartesi/rollups/blob/v0.8.0/.github/workflows/docker.yml#L64-L76